// ng2-image-hover-zoom/src/image-zoom/image-zoom-container.component.js
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
var __metadata = (this && this.__metadata) || function (k, v) {
    if (typeof Reflect === "object" && typeof Reflect.metadata === "function") return Reflect.metadata(k, v);
};
/**
 * Created by craftsvilla on 15/03/17.
 */
import { Component, ElementRef } from '@angular/core';
var ImageZoomContainer = (function () {
    function ImageZoomContainer(_elementRef) {
        this._elementRef = _elementRef;
        this.el = this._elementRef.nativeElement;
    }
    ImageZoomContainer.prototype.setOptions = function (options) {
        this.el.style.backgroundImage = "url(" + options['src'] + ")";
        this.el.style.left = options['left'] + 'px';
        this.el.style.top = options['top'] + 'px';
        this.el.style.width = options['width'] + 'px';
        this.el.style.height = options['height'] + 'px';
    };
    ImageZoomContainer.prototype.setVisibility = function (visible) {
        this.el.style.display = visible ? 'block' : 'none';
    };
    ImageZoomContainer.prototype.setBackgroundPosition = function (position) {
        this.el.style.backgroundPosition = position;
    };
    return ImageZoomContainer;
}());
ImageZoomContainer = __decorate([
    Component({
        selector: 'image-zoom-container',
        template: "",
        styles: ["\n    :host{\n      position: absolute;\n      width: 400px;\n      height: 400px;\n      overflow: hidden;\n      display: none;\n      border: 1px solid #7a7a7a;\n    }\n  "]
    }),
    __metadata("design:paramtypes", [ElementRef])
], ImageZoomContainer);
export { ImageZoomContainer };
