// ng2-zoomable/demo/app.component.js
"use strict";
var __decorate = (this && this.__decorate) || function (decorators, target, key, desc) {
    var c = arguments.length, r = c < 3 ? target : desc === null ? desc = Object.getOwnPropertyDescriptor(target, key) : desc, d;
    if (typeof Reflect === "object" && typeof Reflect.decorate === "function") r = Reflect.decorate(decorators, target, key, desc);
    else for (var i = decorators.length - 1; i >= 0; i--) if (d = decorators[i]) r = (c < 3 ? d(r) : c > 3 ? d(target, key, r) : d(target, key)) || r;
    return c > 3 && r && Object.defineProperty(target, key, r), r;
};
Object.defineProperty(exports, "__esModule", { value: true });
var core_1 = require("@angular/core");
var img1 = require('./assets/thumbnail.jpg');
var img2 = require('./assets/thumbnail2.jpg');
var AppComponent = (function () {
    function AppComponent() {
        this.imageSource1 = img1;
        this.imageSource2 = img2;
    }
    return AppComponent;
}());
AppComponent = __decorate([
    core_1.Component({
        selector: 'my-app',
        styles: ["\n    img {\n      display: block;\n    }\n\n    .img1 {\n      width: 400px;\n      height: 274px;\n    }\n\n    .img2 {\n      margin-top: 20px;\n      width: 388px;\n      height: 580px;\n    }\n  "],
        template: "\n    <img [zoomable]='zoomOptions' [zoomableSrc]='imageSource1' [src]='imageSource1' class='img1'>\n    <img [zoomable]='zoomOptions' [src]='imageSource2' class='img2'>\n  "
    })
], AppComponent);
exports.AppComponent = AppComponent;
